```
[ant:checkstyle] [ERROR] /home/chia7712/kafka/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java:19:1: Disallowed import - com.fasterxml.jackson.core.JsonProcessingException. [ImportControl]
[ant:checkstyle] [ERROR] /home/chia7712/kafka/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java:19:8: Unused import - com.fasterxml.jackson.core.JsonProcessingException. [UnusedImports]
[ant:checkstyle] [ERROR] /home/chia7712/kafka/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java:24:8: Unused import - org.apache.kafka.connect.runtime.Connect. [UnusedImports]
[ant:checkstyle] [ERROR] /home/chia7712/kafka/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java:299:9: 'try' is not followed by whitespace. [WhitespaceAround]
[ant:checkstyle] [ERROR] /home/chia7712/kafka/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java:299:12: '{' is not preceded with whitespace. [WhitespaceAround]
```

+1 after fixing above checkstyle error :)

--- a/connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExampleConnectIntegrationTest.java
+++ b/connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExampleConnectIntegrationTest.java
@@ -16,7 +16,6 @@
  */
 package org.apache.kafka.connect.integration;
 
-import org.apache.kafka.connect.runtime.rest.entities.ConfigInfos;
 import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
 import org.apache.kafka.connect.storage.StringConverter;
 import org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster;
@@ -126,15 +125,15 @@ public class ExampleConnectIntegrationTest {
         connectorHandle.expectedCommits(NUM_RECORDS_PRODUCED);
 
         // validate the intended connector configuration, a config that errors
-        ConfigInfos configInfosError = connect.validateConnectorConfig(SINK_CONNECTOR_CLASS_NAME, props);
-        assertEquals(1, configInfosError.errorCount());
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(SINK_CONNECTOR_CLASS_NAME, props, 1,
+            "Validating connector configuration produced an unexpected number or errors.");
 
         // add missing configuration to make the config valid
         props.put("name", CONNECTOR_NAME);
 
         // validate the intended connector configuration, a valid config
-        ConfigInfos configInfosValid = connect.validateConnectorConfig(SINK_CONNECTOR_CLASS_NAME, props);
-        assertEquals(0, configInfosValid.errorCount());
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(SINK_CONNECTOR_CLASS_NAME, props, 0,
+            "Validating connector configuration produced an unexpected number or errors.");
 
         // start a sink connector
         connect.configureConnector(CONNECTOR_NAME, props);
@@ -187,15 +186,15 @@ public class ExampleConnectIntegrationTest {
         connectorHandle.expectedCommits(NUM_RECORDS_PRODUCED);
 
         // validate the intended connector configuration, a config that errors
-        ConfigInfos configInfosError = connect.validateConnectorConfig(SOURCE_CONNECTOR_CLASS_NAME, props);
-        assertEquals(1, configInfosError.errorCount());
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(SOURCE_CONNECTOR_CLASS_NAME, props, 1,
+            "Validating connector configuration produced an unexpected number or errors.");
 
         // add missing configuration to make the config valid
         props.put("name", CONNECTOR_NAME);
 
         // validate the intended connector configuration, a valid config
-        ConfigInfos configInfosValid = connect.validateConnectorConfig(SOURCE_CONNECTOR_CLASS_NAME, props);
-        assertEquals(0, configInfosValid.errorCount());
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(SOURCE_CONNECTOR_CLASS_NAME, props, 0,
+            "Validating connector configuration produced an unexpected number or errors.");
 
         // start a source connector
         connect.configureConnector(CONNECTOR_NAME, props);

--- a/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java
+++ b/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java
@@ -16,12 +16,10 @@
  */
 package org.apache.kafka.connect.util.clusters;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.kafka.common.utils.Exit;
 import org.apache.kafka.connect.errors.ConnectException;
-import org.apache.kafka.connect.runtime.Connect;
 import org.apache.kafka.connect.runtime.rest.entities.ActiveTopicsInfo;
 import org.apache.kafka.connect.runtime.rest.entities.ConfigInfos;
 import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
@@ -296,7 +294,7 @@ public class EmbeddedConnectCluster {
         String url = endpointForResource(String.format("connector-plugins/%s/config/validate", connClassName));
         String response = putConnectorConfig(url, connConfig);
         ConfigInfos configInfos;
-        try{
+        try {
             configInfos = new ObjectMapper().readValue(response, ConfigInfos.class);
         } catch (IOException e) {
             throw new ConnectException("Unable deserialize response into a ConfigInfos object");

--- a/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectClusterAssertions.java
+++ b/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectClusterAssertions.java
@@ -16,6 +16,7 @@
  */
 package org.apache.kafka.connect.util.clusters;
 
+import java.util.Map;
 import org.apache.kafka.connect.runtime.AbstractStatus;
 import org.apache.kafka.connect.runtime.rest.entities.ActiveTopicsInfo;
 import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
@@ -38,6 +39,7 @@ public class EmbeddedConnectClusterAssertions {
 
     private static final Logger log = LoggerFactory.getLogger(EmbeddedConnectClusterAssertions.class);
     public static final long WORKER_SETUP_DURATION_MS = TimeUnit.SECONDS.toMillis(60);
+    public static final long VALIDATION_DURATION_MS = TimeUnit.SECONDS.toMillis(30);
     private static final long CONNECTOR_SETUP_DURATION_MS = TimeUnit.SECONDS.toMillis(30);
     private static final long CONNECT_INTERNAL_TOPIC_UPDATES_DURATION_MS = TimeUnit.SECONDS.toMillis(60);
 
@@ -91,6 +93,49 @@ public class EmbeddedConnectClusterAssertions {
             return Optional.of(comp.apply(numUp, numWorkers));
         } catch (Exception e) {
             log.error("Could not check active workers.", e);
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Assert that the required number of errors are produced by a connector config validation.
+     *
+     * @param connectorClass the class of the connector to validate
+     * @param connConfig     the intended configuration
+     * @param numErrors      the number of errors expected
+     */
+    public void assertExactlyNumErrorsOnConnectorConfigValidation(String connectorClass, Map<String, String> connConfig,
+        int numErrors, String detailMessage) throws InterruptedException {
+        try {
+            waitForCondition(
+                () -> checkValidationErrors(
+                    connectorClass,
+                    connConfig,
+                    numErrors,
+                    (actual, expected) -> actual == expected
+                ).orElse(false),
+                VALIDATION_DURATION_MS,
+                "Didn't meet the exact requested number of validation errors: " + numErrors);
+        } catch (AssertionError e) {
+            throw new AssertionError(detailMessage, e);
+        }
+    }
+
+    /**
+     * Confirm that the requested number of errors are produced by {@link EmbeddedConnectCluster#validateConnectorConfig}.
+     *
+     * @param connectorClass the class of the connector to validate
+     * @param connConfig     the intended configuration
+     * @param numErrors      the number of errors expected
+     * @return true if exactly {@code numErrors} are produced by the validation; false otherwise
+     */
+    protected Optional<Boolean> checkValidationErrors(String connectorClass, Map<String, String> connConfig,
+        int numErrors, BiFunction<Integer, Integer, Boolean> comp) {
+        try {
+            int numErrorsProduced = connect.validateConnectorConfig(connectorClass, connConfig).errorCount();
+            return Optional.of(comp.apply(numErrorsProduced, numErrors));
+        } catch (Exception e) {
+            log.error("Could not check config validation error count.", e);
             return Optional.empty();
         }
     }

